src/modal/docs/readme.md

```diff
@@ -8,6 +8,7 @@ The `$modal` service has only one method: `open(options)` where available option
 * `scope` - a scope instance to be used for the modal's content (actually the `$modal` service is going to create a child scope of a provided scope). Defaults to `$rootScope`
 * `controller` - a controller for a modal instance - it can initialize scope used by modal. Accepts the "controller-as" syntax in the form 'SomeCtrl as myctrl'; can be injected with `$modalInstance`
 * `controllerAs` - an alternative to the controller-as syntax, matching the API of directive definitions. Requires the `controller` option to be provided as well
+* `bindToController` - when used with `controllerAs` & set to `true`, it will bind the controller properties onto the `$scope` directly
```

Comment:
I think that you understood `bindToController` in the other way around:

https://docs.angularjs.org/api/ng/service/$compile
> *bindToController*: When an isolate scope is used for a component (see above), and controllerAs is used, bindToController: true will allow a component to have its properties bound to the controller, rather than to scope. 

Bind to controller enforces the use of *controllerAs*, and it publishes the controller into the scope as it happens since 1.2. The new point is that all bindings to the directive are made towards the controller instead of towards the scope, but the scope does not dissappear, and you always use in template: ctrl.property.

The problem in angular 1.X is that you need to use the dot notation because you have a high risk (specilly with you work with non expert programmers) to mess it up with double bindings and nested scopes, for instance:

```javascript
Value: {{vm.value}}
<div ng-if="vm.isInputActive()">
  <input ng-model="vm.value">
</div>
```

vs.

```javascript
Value: {{value}}
<div ng-if="vm.isInputActive()">
  <input ng-model="value">
</div>
```

Which will not work as expected according the current implementation proposed in this thread: it will save the value in the scope created by ngIf but not in the directive context.

So, one of two:

1) bindToController does not make sense at all here, there are no bindings to the modal, so better to remove it (my recommendation)

2) ok, we want to bind controller properties to the scope, although it is dangerous, but we should name another name like: controllerPropertiesInScope, to avoid confusions with angular meaning

And according the presented implementation, it does not uses the controller instance at all, it is just a temporary object which instance reference is lost in favor to scope:

https://github.com/angular-ui/bootstrap/commit/8adfc833f710fd1aff5d91950e78362515946767
```javascript
angular.extend(modalScope, ctrlInstance);
```